pulsar-broker/src/test/java/org/apache/pulsar/client/impl/CreateConsumerProducerTest.java

```diff
@@ -0,0 +1,337 @@
+                InetSocketAddress.createUnresolved("127.0.0.1", BROKER_SERVICE_PORT), true);
+        doTestServiceUrlResolve(pulsarClientWithHttpServiceUrl,
+                "http://host1:6651,host2:6651,127.0.0.1:" + WEB_SERVICE_PORT,
+                InetSocketAddress.createUnresolved("127.0.0.1", WEB_SERVICE_PORT), true);
+
+        doTestServiceUrlResolve(pulsarClientWithBinaryServiceUrlDisableQuarantine,
+                "pulsar+ssl://host1:6651,host2:6651,127.0.0.1:" + BROKER_SERVICE_PORT,
+                InetSocketAddress.createUnresolved("127.0.0.1", BROKER_SERVICE_PORT), false);
+        doTestServiceUrlResolve(pulsarClientWithHttpServiceUrlDisableQuarantine,
+                "http://host1:6651,host2:6651,127.0.0.1:" + WEB_SERVICE_PORT,
+                InetSocketAddress.createUnresolved("127.0.0.1", WEB_SERVICE_PORT), false);
+    }
+
+    private void doTestServiceUrlResolve(PulsarClientImpl pulsarClient, String serviceUrl,
+                                         InetSocketAddress healthyAddress, boolean enableQuarantine)
+            throws Exception {
+        LookupService resolver = pulsarClient.getLookup();
+        resolver.updateServiceUrl(serviceUrl);
+        assertEquals(serviceUrl, resolver.getServiceUrl());
+
+        ServiceURI uri;
+        try {
+            uri = ServiceURI.create(serviceUrl);
+        } catch (IllegalArgumentException iae) {
+            log.error("Invalid service-url {} provided {}", serviceUrl, iae.getMessage(), iae);
+            throw new PulsarClientException.InvalidServiceURL(iae);
+        }
+        String[] hosts = uri.getServiceHosts();
+        Set<InetSocketAddress> originAllAddresses = new HashSet<>(hosts.length);
+        for (String host : hosts) {
+            String hostUrl = uri.getServiceScheme() + "://" + host;
+            try {
+                URI hostUri = new URI(hostUrl);
+                originAllAddresses.add(InetSocketAddress.createUnresolved(hostUri.getHost(), hostUri.getPort()));
+            } catch (URISyntaxException e) {
+                log.error("Invalid host provided {}", hostUrl, e);
+                throw new PulsarClientException.InvalidServiceURL(e);
+            }
+        }
+
+        for (int i = 0; i < 10; i++) {
+            assertTrue(originAllAddresses.contains(resolver.resolveHost()));
+        }
+        String topic = "persistent://my-property/my-ns/topic" + UUID.randomUUID();
+        admin.topics().createNonPartitionedTopic(topic);
+        // Create consumers to trigger unhealthy address removal
+        for (int i = 0; i < originAllAddresses.size(); i++) {
+            String subName = "my-sub" + UUID.randomUUID();
+            pulsarClient.newConsumer()
+                    .subscriptionMode(SubscriptionMode.Durable)
+                    .topic(topic).receiverQueueSize(1).subscriptionName(subName)
+                    .subscribeAsync()
+                    .thenAccept(Consumer::closeAsync);
```

Comment:
Please use sync call instead of async call.

Reply:
fixed